File: src/main/java/io/golos/commun4J/model/UnVoteRequest.java
package io.golos.commun4J.model;

import com.memtrip.eos.abi.writer.Abi;
import com.memtrip.eos.abi.writer.NameCompress;
import com.memtrip.eos.abi.writer.StringCompress;

@Abi
public class UnVoteRequest {
    private CommunName voter;
    private CommunName author;
    private String permlink;


    public UnVoteRequest(CommunName voter, CommunName author, String permlink) {
        this.voter = voter;
        this.author = author;
        this.permlink = permlink;

    }

    @NameCompress
    public String getVoter() {
        return voter.getName();
    }


    @NameCompress
    public String getAuthor() {
        return author.getName();
    }

    @StringCompress
    public String getPermlink() {
        return permlink;
    }


    @Override
    public String toString() {
        return "VoteRequest{" +
                "voter=" + voter +
                ", author=" + author +
                ", permlink='" + permlink + '\'' +
                '}';
    }
}

File: src/main/java/io/golos/commun4J/model/UnVoteRequest.kt
package io.golos.commun4J.model

import com.memtrip.eos.abi.writer.Abi
import com.memtrip.eos.abi.writer.ChildCompress
import com.memtrip.eos.abi.writer.NameCompress

@Abi
internal class UnVoteRequest(private val voter: CommunName,
                             private val discussionId: DiscussionId) {


    val getVoter: String
        @NameCompress get() = voter.name


    val getDiscussionId
        @ChildCompress get() = discussionId

    override fun toString(): String {
        return "UnVoteRequest(voter=$voter, discussionId=$discussionId)"
    }
}
